ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -118,6 +119,144 @@ private class JdbcLedgerDao(
+
+  private val SQL_GET_CONFIGURATION_ENTRIES = SQL(
+    "select * from configuration_entries where ledger_offset>={startInclusive} and ledger_offset<{endExclusive} order by ledger_offset asc")
+
+  private def updateCurrentConfiguration(configBytes: Array[Byte])(
+      implicit conn: Connection): Unit = {
+    SQL_UPDATE_CURRENT_CONFIGURATION
+      .on("configuration" -> configBytes)
+      .execute()
+    ()
+  }
+
+  private def selectLedgerConfiguration(implicit conn: Connection) =
+    SQL_SELECT_CURRENT_CONFIGURATION
+      .as(byteArray("configuration").?.single)
+      .flatMap(Configuration.decode(_).toOption)
+
+  override def lookupLedgerConfiguration(): Future[Option[Configuration]] =
+    dbDispatcher.executeSql("lookup configuration")(implicit conn => selectLedgerConfiguration)
+
+  private val configurationEntryParser: RowParser[(Long, ConfigurationEntry)] =
+    (long("ledger_offset") ~
+      str("typ") ~
+      str("submission_id") ~
+      str("participant_id") ~
+      str("rejection_reason")(emptyStringToNullColumn).? ~
+      byteArray("configuration"))
+      .map(flatten)
+      .map {
+        case (offset, typ, submissionId, participantIdRaw, rejectionReason, configBytes) =>
+          val config = Configuration
+            .decode(configBytes)
+            .fold(err => sys.error(s"Failed to decode configuration: $err"), identity)
+          val participantId = LedgerString
+            .fromString(participantIdRaw)
+            .fold(
+              err => sys.error(s"Failed to decode participant id in configuration entry: $err"),
+              identity)
+
+          offset ->
+            (typ match {
+              case "accept" =>
+                ConfigurationEntry.Accepted(
+                  submissionId = submissionId,
+                  participantId = participantId,
+                  configuration = config
+                )
+              case "reject" =>
+                ConfigurationEntry.Rejected(
+                  submissionId = submissionId,
+                  participantId = participantId,
+                  rejectionReason = rejectionReason.getOrElse("<missing reason>"),
+                  proposedConfiguration = config
+                )
+
+              case _ =>
+                sys.error(s"getConfigurationEntries: Unknown configuration entry type: $typ")
+            })
+      }
+
+  override def getConfigurationEntries(
+      startInclusive: Long,
+      endExclusive: Long): Source[(Long, ConfigurationEntry), NotUsed] =
+    paginatingStream(
+      startInclusive,
+      endExclusive,
+      PageSize,
+      (startI, endE) => {
+        Source
+          .fromFuture(dbDispatcher.executeSql(s"load ledger entries [$startI, $endE[") {
+            implicit conn =>
+              SQL_GET_CONFIGURATION_ENTRIES
+                .on("startInclusive" -> startI, "endExclusive" -> endE)
+                .as(configurationEntryParser.*)
+          })
+          .flatMapConcat(Source(_))
```

Comment:
We should probably use `mapConcat(identity)` instead of materializing a sub source (also for `getLedgerEntries`).